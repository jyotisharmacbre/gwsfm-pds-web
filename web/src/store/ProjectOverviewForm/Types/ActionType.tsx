export enum ActionType {
	PROJECT_OVERVIEW_FORM_ADD_SUCCESS = 'PROJECT_OVERVIEW_FORM_ADD_SUCCESS',
	PROJECT_OVERVIEW_FORM_EDIT_SUCCESS = 'PROJECT_OVERVIEW_FORM_EDIT_SUCCESS',
	PROJECT_OVERVIEW_FORM_ERROR = 'PROJECT_OVERVIEW_FORM_ERROR',
	GET_ADDITIONALS_DETAILS_SUCCESS = 'GET_ADDITIONALS_DETAILS_SUCCESS',
	GET_ADDITIONALS_DETAILS_ERROR = 'GET_ADDITIONALS_DETAILS_ERROR',
	RESET_PROJECT_OVERVIEW_STATE = 'RESET_PROJECT_OVERVIEW_STATE',
	BIND_PROJECT_APPROVAL_INITIAL_DATA = 'BIND_PROJECT_APPROVAL_INITIAL_DATA',
	GET_PROJECT_ACTIVITIES_SUCCESS = 'GET_PROJECT_ACTIVITIES_SUCCESS',
	GET_PROJECT_ACTIVITIES_ERROR = 'GET_PROJECT_ACTIVITIES_ERROR',
	RESET_PROJECT_OVERVIEW_NOTIFIER = 'RESET_PROJECT_OVERVIEW_NOTIFIER',
	SET_LOADING_TRUE = 'SET_LOADING_TRUE'
}
