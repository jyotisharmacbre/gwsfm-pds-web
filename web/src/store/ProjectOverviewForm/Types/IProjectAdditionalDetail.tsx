export interface IProjectAdditionalDetail {
  projectAddDetailId: string;
  projectId: string;
  mainContractor: string;
  enquiryReceivedFrom: string;
  potentialCustomer: string;
  enquiryTypeId: string;
  creditCheckResult: string;
  siteAddress: string;
  cdmNotifiable: boolean;
  formOfContract: string;
  retention: string;
  liquidatedDamages: string;
  insurance: string;
  workTypeId: string;
  commenceDate: string;
  completionDate: string;
  milestones: string;
  firstValuationDate: string;
  finalAccountDate: string;
  valuationIntervals: string;
  paymentTerms: string;
  isProjectLive: boolean;
  comments: string;
  authorizedByHop: string;
  budget: string;
  authorizedBy: string;
  authorizedBySecond: string;
  authorizedByThird: string;
}
