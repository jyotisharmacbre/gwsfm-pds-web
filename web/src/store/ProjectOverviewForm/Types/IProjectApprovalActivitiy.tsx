export interface IProjectApprovalActivitiy {
  ApprovalActivityId: string;
  ProjectId: string;
  ApproverTypeId: string;
  ApproverUserId: string;
  Query: string;
  ActivityType: string;
}
