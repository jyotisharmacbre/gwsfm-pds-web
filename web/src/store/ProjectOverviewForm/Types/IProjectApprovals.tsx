export interface IProjectApprovals {
      projectApprovalId: string;
      projectId: string;
      projectApprovalRange: string;
      approverType: string;
      approvalStatus: string;
      userId: string;
  }