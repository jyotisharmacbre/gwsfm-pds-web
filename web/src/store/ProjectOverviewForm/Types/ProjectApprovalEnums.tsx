export enum ProjectApprovalRange {
    Authorised_By = 1,
    Authorised_By_Up_To_CurrencySymbol_100K,
    Authorised_By_Up_To_CurrencySymbol_250K,
    Authorised_By_Up_To_CurrencySymbol_1_Million,
    Authorised_By_Up_To_CurrencySymbol_3_Million,
    Over_CurrencySymbol_3_Million_Authority,
    Over_CurrencySymbol_4_And_half_Million
}

export enum ProjectApproverType {
    HOP = 1,
    AGMOrAD,
    ComM,
    BUL,
    DPD,
    DMD,
    DofP,
    PComM,
    COOOrUKFD,
    CEOOrGroupFD,
    CBRERegulations
}

export enum ProjectSignOffStatus {
    Draft = 1,
    Approved,
    Pending,
    Response_Awaited
}
