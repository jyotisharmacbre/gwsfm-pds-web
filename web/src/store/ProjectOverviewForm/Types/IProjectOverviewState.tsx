import { IProjectAdditionalDetail } from './IProjectAdditionalDetail';
import Notify from '../../../enums/Notify';
import EventType from '../../../enums/EventType';

export interface IProjectOverviewState {
  form: IProjectAdditionalDetail;
  error: string | null;
  loading: boolean;
  notify: Notify;
  event: EventType;
}
