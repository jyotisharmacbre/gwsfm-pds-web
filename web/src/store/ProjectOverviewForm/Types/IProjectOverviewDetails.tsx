import { IProjectAdditionalDetail } from "./IProjectAdditionalDetail";
import { IProjectApprovals } from "./IProjectApprovals";

export interface IProjectOverviewDetails {
    projectId: string;
    projectAdditionalDetail: IProjectAdditionalDetail;
    projectApprovals: Array<IProjectApprovals>;
  }
  