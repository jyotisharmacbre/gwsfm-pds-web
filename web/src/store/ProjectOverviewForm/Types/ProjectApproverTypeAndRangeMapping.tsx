import { ProjectApproverType, ProjectApprovalRange } from "./ProjectApprovalEnums";

interface IApproverTypeAndRangeMapping {
    type: number,
    range: number
}

export const ProjectApproverTypeAndRangeMapping: Array<IApproverTypeAndRangeMapping> = [
    { type: ProjectApproverType.HOP, range: ProjectApprovalRange.Authorised_By },
    { type: ProjectApproverType.AGMOrAD, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_100K },
    { type: ProjectApproverType.ComM, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_250K },
    { type: ProjectApproverType.BUL, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_250K },
    { type: ProjectApproverType.DPD, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_1_Million },
    { type: ProjectApproverType.DMD, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_1_Million },
    { type: ProjectApproverType.DofP, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_3_Million },
    { type: ProjectApproverType.PComM, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_3_Million },
    { type: ProjectApproverType.COOOrUKFD, range: ProjectApprovalRange.Authorised_By_Up_To_CurrencySymbol_3_Million },
    { type: ProjectApproverType.CEOOrGroupFD, range: ProjectApprovalRange.Over_CurrencySymbol_3_Million_Authority },
    { type: ProjectApproverType.CBRERegulations, range: ProjectApprovalRange.Over_CurrencySymbol_4_And_half_Million },
]
