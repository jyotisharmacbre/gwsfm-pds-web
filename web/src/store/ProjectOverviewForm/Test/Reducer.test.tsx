import {ActionType} from '../Types/ActionType';
import projectOverviewFormReducer from '../Reducer';
import nock from 'nock';
import { baseURL } from '../../../client/client';
import { initialState } from './Reducertestdata';

  nock(baseURL)
  .post('/api/Projects/additionalDetails')
  .reply(200, "Project overview form added successfully");
  nock(baseURL)
  .put('/api/Projects/additionalDetails')
  .reply(200, "Project overview form updated successfully");
  nock(baseURL)
  .get('/api/Projects/4d27e2e1-843d-435a-b27c-03dca70ce232/additionalDetails')
  .reply(200, "Project overview data fetched successfully");
  nock(baseURL)
  .put('/api/Projects/4d27e2e1-843d-435a-b27c-03dca70ce232/bidlost')
  .reply(200, "Project overview status updated successfully");
  nock(baseURL)
  .put('/api/Projects/4d27e2e1-843d-435a-b27c-03dca70ce232/onHold')
  .reply(200, "Project overview status updated successfully");
  nock(baseURL)
  .put('/api/Projects/4d27e2e1-843d-435a-b27c-03dca70ce232/reactivate')
  .reply(200, "Project overview status reactivated successfully");

  describe('Project overview Reducer', () => {

      it('should handle add project overview data ', () => {
        const projectOverviewFormAddAction: any = {
          type: ActionType.PROJECT_OVERVIEW_FORM_ADD_SUCCESS
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormAddAction)
        ).toMatchSnapshot();
      });  
      it('should handle error while add project overview data ', () => {
        const projectOverviewFormErrorAction: any = {
          type: ActionType.PROJECT_OVERVIEW_FORM_ERROR
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormErrorAction)
        ).toMatchSnapshot();
      });  
      it('should handle edit  project overview data ', () => {
        const projectOverviewFormEditAction: any = {
          type: ActionType.PROJECT_OVERVIEW_FORM_EDIT_SUCCESS
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormEditAction)
        ).toMatchSnapshot();
      }); 
      it('should handle on hold project status change event ', () => {
        const projectOverviewFormOnHoldAction: any = {
          type: ActionType.CHANGE_PROJECT_STATUS_TO_ON_HOLD_SUCCESS
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormOnHoldAction)
        ).toMatchSnapshot();
      }); 
      it('should handle on hold error during change of project status ', () => {
        const projectOverviewFormOnHoldErrorAction: any = {
          type: ActionType.CHANGE_PROJECT_STATUS_TO_ON_HOLD_ERROR
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormOnHoldErrorAction)
        ).toMatchSnapshot();
      }); 
      it('should handle bid lost project status change event ', () => {
        const projectOverviewFormBidLostAction: any = {
          type: ActionType.CHANGE_PROJECT_STATUS_TO_BID_LOST_SUCCESS
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormBidLostAction)
        ).toMatchSnapshot();
      }); 
      it('should handle bid lost error during change of project status ', () => {
        const projectOverviewFormBidLostErrorAction: any = {
          type: ActionType.CHANGE_PROJECT_STATUS_TO_ON_HOLD_ERROR
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormBidLostErrorAction)
        ).toMatchSnapshot();
      }); 
      it('should handle reactivate project event ', () => {
        const projectOverviewFormReactivateAction: any = {
          type: ActionType.REACTIVATE_PROJECT_SUCCESS
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormReactivateAction)
        ).toMatchSnapshot();
      }); 
      it('should handle reactivate error event ', () => {
        const projectOverviewFormReactivateErrorAction: any = {
          type: ActionType.REACTIVATE_PROJECT_ERROR
        };
        expect(
            projectOverviewFormReducer(initialState, projectOverviewFormReactivateErrorAction)
        ).toMatchSnapshot();
      });
   
});
