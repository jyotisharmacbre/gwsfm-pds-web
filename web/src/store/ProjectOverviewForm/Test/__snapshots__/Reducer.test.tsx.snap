// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project overview Reducer should handle add project overview data  1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`Project overview Reducer should handle bid lost error during change of project status  1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 2,
}
`;

exports[`Project overview Reducer should handle bid lost project status change event  1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`Project overview Reducer should handle edit  project overview data  1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`Project overview Reducer should handle error while add project overview data  1`] = `
Object {
  "error": undefined,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 2,
}
`;

exports[`Project overview Reducer should handle on hold error during change of project status  1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 2,
}
`;

exports[`Project overview Reducer should handle on hold project status change event  1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`Project overview Reducer should handle reactivate error event  1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 2,
}
`;

exports[`Project overview Reducer should handle reactivate project event  1`] = `
Object {
  "error": null,
  "event": 0,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "2019-12-17T11:22:25.498Z",
    "comments": "",
    "completionDate": "2019-12-17T11:22:25.498Z",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": -1,
    "finalAccountDate": "2019-12-17T11:22:25.498Z",
    "firstValuationDate": "2019-12-17T11:22:25.498Z",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "otherMainContractor": "",
    "otherPotentialCustomer": "",
    "paymentTerms": "",
    "potentialCustomer": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;
